netdisco/discoverables/songpal.py

```diff
@@ -0,0 +1,18 @@
+"""Discover Songpal devices."""
+from . import SSDPDiscoverable
+
+
+class Discoverable(SSDPDiscoverable):
+    """Support for Songpal devices.
+    Supported devices: http://vssupport.sony.net/en_ww/device.html."""
+
+    def get_entries(self):
+        """Get all the Songpal devices."""
+        return self.find_by_st(
+            "urn:schemas-sony-com:service:ScalarWebAPI:1")
+
+    def info_from_entry(self, entry):
```

Comment:
No need to add this method if you are not going to make changes.

The SSDP description has been downloaded so it wouldn't be much extra work to extract the port? I think that it's fine to have that in netdisco if it's really just extracting it out of the description.

Reply:
Indeed, I somehow missed that (looks like no devices extract extra information out from them). Is it okay to have a duplicated, friendlier name even when passing a complete data dictionary for the API alongside?

```
[{'endpoint': 'http://192.168.250.241:10000/sony',
  'host': '192.168.250.241',
  'model_name': 'HT-XT3',
  'model_number': 'BAR-2015',
  'name': 'HT-XT3',
  'port': 52323,
  'scalarwebapi': {'X_ScalarWebAPI_BaseURL': 'http://192.168.250.241:10000/sony',
                   'X_ScalarWebAPI_ServiceList': {'X_ScalarWebAPI_ServiceType': ['guide',
                                                                                 'system',
                                                                                 'audio',
                                                                                 'avContent']},
                   'X_ScalarWebAPI_Version': '1.0'},
  'serial': None,
  'ssdp_description': 'http://192.168.250.241:52323/dmr.xml'}]
```

Here `endpoint` is extracted from the `scalarwebapi` for easier usage.

Reply:
Do you think that it makes sense to overwrite the port? As the 52323 seems to be only used for running the SSDP server.

Reply:
I don't think it makes sense to rewrite it, who knows for what it can be useful and that endpoint I'm using still requires a path to be useful and I have no idea if it is static between different types of devices.

These devices also offer yet another, more lower-level API (X_CIS_DeviceInfo, maybe https://shop.kindermann.de/erp/KCO/avs/3/3005/3005000168/01_Anleitungen+Doku/Steuerungsprotokoll_1.pdf), which could be useful for some people. https://github.com/aschaeffer/sony-av-indicator has a potential implementation, but I have not tested it out (just leaving these links here in case someone comes searching).